### PR TITLE
Challenge completed

### PR DESCRIPTION
**Pull request by Bondesvick**
Please review my solution and give me necessary suggestions
Thanks

### CODE REVIEW

File: src/word-search/index.js [hunk 1]
```diff
@@ -1,3 +1,74 @@
...
+    if (words[row][col + 1] == word[matchIndex + 1] &&
+      list[row.toString() + (col + 1).toString()] !== null) {
+
+      crawler(row, col + 1, matchIndex + 1)
+
+    }
+  }
+
+  return match
+}
```
Comment (flinchy) [inline src/word-search/index.js:73]: great job! add comments to your code where necessary to explain what the code does, best practices

File: src/count-islands/index.js [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+      crawler(row, col - 1)
+
+      } if (grid[row][col + 1] == 1 &&
+        list[row.toString() + (col + 1).toString()] !== null) {
+          crawler(row, col + 1)
+      }
+  }
+
+  return count;
+}
```
Comment (flinchy) [inline src/count-islands/index.js:46]: great job! add comments to your code where necessary to explain what the code does, best practices

File: src/word-search/index.js (hunk 1 above)
Comment (Bondesvick) [reply src/word-search/index.js:101]: Comments added

File: src/word-search/index.js [hunk 3]
```diff
@@ -1,3 +1,102 @@
...
+          match = true
+          return
+        }
+      }
+    })
+  });
+  //#endregion
+
+  //#region - A dfs function that handles searching adjacent characters that matches with the next letter in the word
+  function crawler(row, col, matchIndex) {
```
Comment (flinchy) [inline src/word-search/index.js:48]: Move this crawler function out of the wordSearch() function, a function should have a single responsibility, the wordSearch function is getting too complex so many things going on in there

File: src/count-islands/index.js [hunk 4]
```diff
@@ -1,3 +1,47 @@
...
+    element.forEach((item, childIndex) => {
+      if (item == 1 &&
+        list[rootIndex.toString() + childIndex.toString()] !== null) {
+        count++
+        crawler(rootIndex,childIndex)
+      }
+    })
+  });
+
+  function crawler(row, col) {
```
Comment (flinchy) [inline src/count-islands/index.js:21]: move this crawler function out of the countIsLand() function

File: src/word-search/index.js (hunk 3 above)
Comment (Bondesvick) [reply src/word-search/index.js:48]: Oh, I get your point, didn't think it mattered. I'll fix it right away.

Comment (flinchy) [reply src/word-search/index.js:48]: That way your code is easily maintainable and reusable also

Comment (Bondesvick) [reply src/word-search/index.js]: done

File: src/count-islands/index.js (hunk 4 above)
Comment (Bondesvick) [reply src/count-islands/index.js]: done

**Pull request merged**